src/Contract/Address.purs

```diff
@@ -180,9 +189,12 @@ getWalletAddressesWithNetworkTag = do
 -- | Throws on `Promise` rejection by wallet, returns `Nothing` if no collateral
 -- | is available.
 getWalletCollateral
-  :: forall (r :: Row Type). Contract r (Maybe (Array TransactionUnspentOutput))
-getWalletCollateral = do
-  mtxUnspentOutput <- wrapContract QueryM.getWalletCollateral
+  :: forall (r :: Row Type)
+   . Maybe Value
+  -> Contract r (Maybe (Array TransactionUnspentOutput))
+getWalletCollateral amount = do
```

Comment:
We should **require** an amount, according to CIP-30 (please remove `Maybe`). The amount value should be provided as a `Coin`.

> The amount parameter is required, specified as a string (BigNumber) or a number, 

We should use a string in FFI

> and the maximum allowed value must be agreed to be something like 5 ADA

This is of course a bad wording in the standard, but we should also enforce the 5 ADA maximum.

Reply:
But this is not CIP-30 interface, but our internal interface, there Maybe is useful.

Reply:
OK I agree. Please make sure the semantics for `Nothing` are well described in the doc-comment

Reply:
You were right, we can just use `mempty` as Nothing here.